doc/feature/transferFunctions.md

```diff
@@ -0,0 +1,238 @@
+        bool isEmpty() const;
+
+        /** Return the range [m', M'] to be used in texture lookups.
+         *
+         *  This range ensures that the mininum x of the control points maps to
+         *  the middle of the first texel (1/2t) and maximum x of the control
+         *  points maps to (1 - 1/2t), where t in the texture size.
+         */
+        getTextureMappingRange(size_t textureSize, float& min, float& max);
+
+        servus::Serializable& getSerializable();
+
+        const servus::Serializable& getSerializable() const;
+
+        /* Other functions that may be useful */
+
+        /**
+         * Set the control points from a list of map of x values to RGBA tuples.
+         *
+         * It could be a contructor as well
+         */
+        void setControlPoints( const std::map< float, vmml::Vector4f >& points );
+
+        /**
+         * Set the control points from a list of RGBA tuples and a range.
+         *
+         * The number of control points will be points.size() and the x values
+         * will be the linear interpolation of [min, max] at 1/points.size()
+         * intervals.
+         *
+         * It could be a contructor as well
+         */
+        void setControlPoints( const std::vector< vmml::Vector4f >& points,
+                               float min, float max );
+
+        /**
+         * Readjust the x values of all control points to find the new min, max
+         * range.
+         *
+         * Each channel will be adjusted preserving its relative range to the
+         * max and min x-values for all channels.
+         */
+        void adjustPoints(float min, float max);
+    }
+
+## Issues
+
+### 1: Why a transfer function is not described with a lookup table ?
+
+_Resolved: Yes_
+
+Because a lookup table is a predefined sampling of the transfer function. On
+the other hand a piece wise funtion with control points can be more precise
+for transfer functions with high frecuencies and more light weight for simple
+linear functions. Furthermore, depending on the rendering engine a lookup table
+may not be needed or may not be sufficient (e.g. pre-integrated volume rendering)
+The C++ API still provides a function to easily convert the message into a
+lookup table as this is needed by OpenGL.
+
+### 2: Why the colormap uses independent control point lists for each channel ?
+
+Colormap editors typically have RGBA control poitns, however, advanced transfer
+function editors have separate curves for each channel. An event with separate
+control point lists easily allows the latter, while it's still trivial to use
+it for the former.
+
+_Resolved: Yes_
+
+### 3: ZeroEQ doesn't allow to subscribe multiple instances of a single
+       serializable type neither in the PUB/SUB nor the HTTP interfaces. How
+       does this specification allow the support of multiple colormaps per
+       applications (e.g. as RTNeuron supports in its Python API)
+
+_Resolved: No_
```

Comment:
I've been discussing it with Daniel and I've come up with an idea that I'd like to share with you tomorrow.